lib/net.js

```diff
@@ -1536,6 +1536,11 @@ function onconnection(err, clientHandle) {
     return;
   }
 
+  if (self.listenerCount('connection') === 0) {
```

Comment:
I have a feeling this might be expensive for _all_ incoming connections. Instead, you can check the return value of `emit`: https://nodejs.org/api/events.html#events_emitter_emit_eventname_args.